src/CoreServices/FSEvents.cs

```diff
@@ -199,6 +199,30 @@ public FSEventStream (string [] pathsToWatch, TimeSpan latency, FSEventStreamCre
 		{
 		}
 
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern IntPtr FSEventStreamCreateRelativeToDevice (IntPtr allocator, FSEventStreamCallback callback, 
+		IntPtr context, ulong deviceToWatch, IntPtr pathsToWatchRelativeToDevice, 
+		ulong sinceWhen, double latency, FSEventStreamCreateFlags flags);
+
+		public FSEventStream (CFAllocator allocator, ulong deviceToWatch, NSArray pathsToWatchRelativeToDevice, 
+			ulong sinceWhen, TimeSpan latency, FSEventStreamCreateFlags flags)
+		{
+			if (pathsToWatchRelativeToDevice == null) {
+				throw new ArgumentNullException ("pathsToWatchRelativeToDevice");
+			}
+
+			eventsCallback = new FSEventStreamCallback (EventsCallback);
+
+			handle =  FSEventStreamCreateRelativeToDevice (
+				allocator ==  null ? IntPtr.Zero : allocator.Handle,
```

Comment:
you can use the extension method `GetHandle ()` here, e.g.
```csharp
allocator.GetHandle (),
```
This will do the null check for you (and it's easier to read)